File: JM/Brute-force/4. carpet.java
//카펫
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        //yellow = (w-2)*(y-2)
        //brown = w*h
        int w_plus_h = (brown+4)/2;
        int w=w_plus_h-3;
        while(true){
            int h=w_plus_h-w;
            if(w*h==brown+yellow){
                answer[0] = w;
                answer[1] = h; break;
            }
            w--;
        }
        return answer;
    }
}

File: HS/04_Search/4_carpet.js
function solution(brown, yellow) {
  for (let i = 1; i <= brown / 2; i++) {
    if (yellow % i == 0) {
      //약수
      width = yellow / i + 2;
      if (2 * (width + i) == brown) {
        return [width, i + 2];
      }
    }
  }
}
